File: app/src/main/java/com/pkfcooperparry/template/ui/form/page/tab/TabTitlesPagerAdapter.java
package com.pkfcooperparry.template.ui.form.page.tab;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.pkfcooperparry.template.R;
import com.pkfcooperparry.template.data.model.db.visit.Tab;

import java.util.List;

public class TabTitlesPagerAdapter extends RecyclerView.Adapter<TabTitlesPagerAdapter.ViewHolder> {

    Context context;
    int selectedItem = 0;
    private List<Tab> tabs;
    private LayoutInflater inflater;
    private ItemClickListener itemClickListener;

    // data is passed into the constructor
    public TabTitlesPagerAdapter(Context context, ItemClickListener itemClickListener, List<Tab> tabs) {
        this.inflater = LayoutInflater.from(context);
        this.itemClickListener = itemClickListener;
        this.context = context;
        this.tabs = tabs;
    }

    // inflates the row layout from xml when needed
    @Override
    @NonNull
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.row_tab, parent, false);
        return new ViewHolder(view);
    }

    // binds the data to the view and textview in each row
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Tab tab = tabs.get(position);
        if (position == selectedItem) {
            holder.button.setBackgroundColor(ContextCompat.getColor(context, R.color.colorAccent));
            holder.button.setTextColor(ContextCompat.getColor(context, android.R.color.white));
        } else {
            holder.button.setBackgroundColor(ContextCompat.getColor(context, android.R.color.white));
            holder.button.setTextColor(ContextCompat.getColor(context, R.color.colorLightTextColour));

        }
        holder.button.setText(tab.getTabName());
    }

    // total number of rows
    @Override
    public int getItemCount() {
        return tabs.size();
    }

    public void updateSelected(int selectedItem) {
        int temp = this.selectedItem;
        this.selectedItem = selectedItem;
        notifyItemChanged(temp);
        notifyItemChanged(selectedItem);
    }

    public void setClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }


    public interface ItemClickListener {
        void onItemClick(View view, int position);
    }

    // stores and recycles views as they are scrolled off screen
    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        Button button;

        ViewHolder(View itemView) {
            super(itemView);
            button = itemView.findViewById(R.id.button);
            button.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (itemClickListener != null) {
                itemClickListener.onItemClick(view, getAdapterPosition());
            }
        }
    }
}

File: app/src/main/java/com/televantou/template/ui/form/page/tab/TabTitlesPagerAdapter.kt
package com.televantou.template.ui.form.page.tab

import android.content.Context
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button

import com.televantou.template.R
import com.televantou.template.data.model.db.visit.Tab

class TabTitlesPagerAdapter// data is passed into the constructor
(internal var context: Context, private var itemClickListener: ItemClickListener?, private val tabs: List<Tab>) : RecyclerView.Adapter<TabTitlesPagerAdapter.ViewHolder>() {
    internal var selectedItem = 0
    private val inflater: LayoutInflater

    init {
        this.inflater = LayoutInflater.from(context)
    }

    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = inflater.inflate(R.layout.row_tab, parent, false)
        return ViewHolder(view)
    }

    // binds the data to the view and textview in each row
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val tab = tabs[position]
        if (position == selectedItem) {
            holder.button.setBackgroundColor(ContextCompat.getColor(context, R.color.colorAccent))
            holder.button.setTextColor(ContextCompat.getColor(context, android.R.color.white))
        } else {
            holder.button.setBackgroundColor(ContextCompat.getColor(context, android.R.color.white))
            holder.button.setTextColor(ContextCompat.getColor(context, R.color.colorLightTextColour))

        }
        holder.button.text = tab.tabName
    }

    // total number of rows
    override fun getItemCount(): Int {
        return tabs.size
    }

    fun updateSelected(selectedItem: Int) {
        val temp = this.selectedItem
        this.selectedItem = selectedItem
        notifyItemChanged(temp)
        notifyItemChanged(selectedItem)
    }

    fun setClickListener(itemClickListener: ItemClickListener) {
        this.itemClickListener = itemClickListener
    }


    interface ItemClickListener {
        fun onItemClick(view: View, position: Int)
    }

    // stores and recycles views as they are scrolled off screen
    inner class ViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        internal var button: Button

        init {
            button = itemView.findViewById(R.id.button)
            button.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            if (itemClickListener != null) {
                itemClickListener!!.onItemClick(view, adapterPosition)
            }
        }
    }
}